internal/gatewayapi/resource/resource.go

```diff
@@ -29,13 +29,27 @@ type (
 	InfraIRMap map[string]*ir.Infra
 )
 
+// LoadResources extends Resources to allow holding
+// multiple EnvoyProxies and GatewayClasses.
+//
+// This structure is only used for marshalling/unmarshalling
+// purposes and is not used by the translator.
+type LoadResources struct {
+	// Resources have some fields that will be reused, except:
```

Comment:
the Gateway API layer relies on on `EnvoyProxyForGatewayClass` and `EnvoyProxiesForGateways` to be set properly, is that being done in the file provider?